### PR TITLE
separate vertex and edge

### PR DESCRIPTION
**Pull request by critical27**
1. Separate vertex and edge by different prefix
2. Add an upgrade tool skeleton which can use both NebulaKeyUtils of 1.0 and 2.0. (NebulaKeyUtilsV1 should only use in upgrader, and support only basic method)

No business logic is changed in this PR. Some code need to refine later, e.g., CompactionFilter and PrefixBloomFilter.

### CODE REVIEW

**Comment by bright-starry-sky** [comment]
Very timely PR. Good job.

**Comment by panda-sheep** [comment]
`DbDumper `should be retained
The DbDumper tool not only uses statistics, but also exports data.

**Comment by critical27** [comment]
> `DbDumper `should be retained
> The DbDumper tool not only uses statistics, but also exports data.

Which DbDumper? I mean `src/tools/db-dump/DbDumper.cpp`

File: src/storage/test/StatisTaskTest.cpp [hunk 1]
```diff
@@ -305,53 +305,63 @@ TEST_F(StatisTaskTest, StatisTagAndEdgeData) {
...
-                if (NebulaKeyUtils::isVertex(spaceVidLen, key)) {
-                    auto vId = NebulaKeyUtils::getVertexId(spaceVidLen, key).str();
-                    auto tagId = NebulaKeyUtils::getTagId(spaceVidLen, key);
-                    auto it = tagsVertices.find(tagId);
-                    if (it == tagsVertices.end()) {
-                        // Invalid data
-                        iter->next();
-                        continue;
-                    }
+                CHECK(NebulaKeyUtils::isVertex(spaceVidLen, key));
```
Comment (panda-sheep) [inline src/storage/test/StatisTaskTest.cpp:308]: Can this check be removed?

File: src/storage/test/StatisTaskTest.cpp [hunk 2]
```diff
@@ -305,53 +305,63 @@ TEST_F(StatisTaskTest, StatisTagAndEdgeData) {
...
-                        spaceEdges++;
-                        edgetypeEdges[edgeType] += 1;
-                        lastSrcVertexId = source;
-                        lastEdgeType  = edgeType;
-                        lastRank = ranking;
-                        lastDstVertexId = destination;
-                    }
+            while (iter && iter->valid()) {
+                auto key = iter->key();
+                CHECK(NebulaKeyUtils::isEdge(spaceVidLen, key));
```
Comment (panda-sheep) [inline src/storage/test/StatisTaskTest.cpp:342]: ditto

**Review by panda-sheep — APPROVED** [review]
Good job

**Review by bright-starry-sky — APPROVED** [review]
LGTM

**Pull request merged**